sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/UnnecessaryImportsCheck.kt

```diff
@@ -170,8 +171,8 @@ class UnnecessaryImportsCheck : AbstractCheck() {
                     ?.resultingDescriptor
                     ?.fqNameOrNull()?.let { listOf(it) }
                     ?: (propDelegate.parent as? KtProperty)?.let { ktProperty ->
-                        (bindingContext.get(BindingContext.DECLARATION_TO_DESCRIPTOR, ktProperty) as? PropertyDescriptor)?.let { propDescriptor ->
-                            propDescriptor
+                        (bindingContext.get(BindingContext.DECLARATION_TO_DESCRIPTOR, ktProperty) as? VariableDescriptorWithAccessors)?.let { descriptor ->
+                            descriptor
                                 .accessors
                                 .mapNotNull {
                                     bindingContext.get(BindingContext.DELEGATED_PROPERTY_RESOLVED_CALL, it)
```

Comment:
Not caused by your commit, but I think we should reduce the complexity of this function in general.